server/lib/gutenbach

```diff
@@ -206,26 +206,35 @@ elsif ($arguments{copies} == $COPIES_SENTINEL) {
       print ZEPHYR "  enqueueing YouTube video /watch?v=$1\n";
       enqueue_url("https://www.youtube.com/watch?v=$1");
     }
-  } elsif (is_youtube_url($filepath)) {
-    # YouTube URLs are resolved by the youtube-dl command.
-
-    # Launch youtube-dl, first just to get the title...
-    $pid = open(YTDL_TITLE, "-|", "/usr/local/bin/youtube-dl", "--no-playlist", "--get-title", $filepath) or die "Unable to invoke youtube-dl";
-    $title = <YTDL_TITLE>;
-    chomp $title;
-
-    # ...then to get the URL.
-    $pid = open(YTDL, "-|", "/usr/local/bin/youtube-dl", "--format", "bestaudio/best", "--no-playlist", "--get-url", $filepath) or die "Unable to invoke youtube-dl";
-    print ZEPHYR "YouTube video $filepath\n$title";
-    $status .= " YouTube video $filepath. $title.";
-
-    # Finally, pass the URL to vlc as a filename.
-    $filepath = <YTDL>;
-    chomp $filepath;
-  } else { # Doesn't appear to be a YouTube URL.
-    print STDERR "Resolved external reference to $filepath\n";
-    print ZEPHYR "$filepath\n";
-    $status .= " External: $filepath\n";
+  } else {
+    # Returns 1 on success, 0 on failure
+    sub youtubedl_get_title_and_url {
+      # Launch youtube-dl, first just to get the title...
+      $pid = open(YTDL_TITLE, "-|", "/usr/local/bin/youtube-dl", "--no-playlist", "--get-title", $filepath) or return 0;
+      $title = <YTDL_TITLE>;
+      close(YTDL_TITLE);
+      return 0 if $?;
+      chomp $title;
+
+      # ...then to get the URL.
+      $pid = open(YTDL, "-|", "/usr/local/bin/youtube-dl", "--format", "bestaudio/best", "--no-playlist", "--get-url", $filepath) or return 0;
+      print ZEPHYR "YouTube video $filepath\n$title";
+      $status .= " YouTube video $filepath. $title.";
+
+      # Finally, pass the URL to vlc as a filename.
+      $filepath = <YTDL>;
+      close(YTDL);
+      return 0 if $?;
+      chomp $filepath;
+
+      return 1;
+    }
+
+    if (!youtubedl_get_title_and_url()) {
+      print STDERR "Unable to invoke youtube-dl; Resolved external reference to $filepath\n";
```

Comment:
I don’t think this fallback case is needed, because `youtube-dl` already resolves a direct URL of an audio file to itself.

Reply:
I see. I was unsure of what the failback case was meant to handle so I left it in there just in case. I will remove it and change change the "return 0"s to "die"s.